starknet_py/tests/e2e/fixtures/utils.py

```diff
@@ -0,0 +1,149 @@
+    for item in items:
+        runs_on_testnet = "run_on_testnet" in item.keywords
+        runs_on_devnet = "run_on_devnet" in item.keywords
+        should_not_run = (runs_on_devnet and not run_devnet) or (
+            runs_on_testnet and not run_testnet
+        )
+        if should_not_run:
+            item.add_marker(pytest.mark.skip())
+
+
+def get_available_port() -> int:
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as sock:
+        sock.bind(("", 0))
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        return sock.getsockname()[1]
+
+
+def start_devnet():
+    devnet_port = get_available_port()
+
+    command = [
+        "poetry",
+        "run",
+        "starknet-devnet",
+        "--host",
+        "localhost",
+        "--port",
+        str(devnet_port),
+        "--accounts",  # deploys specified number of accounts
+        str(1),
+        "--seed",  # generates same accounts each time
+        str(1),
+    ]
+    # pylint: disable=consider-using-with
+    proc = subprocess.Popen(command)
+    time.sleep(5)
+    return devnet_port, proc
+
+
+@pytest.fixture(scope="module")
+def run_devnet() -> Generator[str, None, None]:
+    """
+    Runs devnet instance once per module and returns it's address
+    """
+    devnet_port, proc = start_devnet()
+    yield f"http://localhost:{devnet_port}"
+    proc.kill()
+
+
+@pytest.fixture(scope="module")
+def network(pytestconfig, run_devnet: str) -> str:
+    """
+    Returns network address depending on the --net parameter
+    """
+    net = pytestconfig.getoption("--net")
+    net_address = {
+        "devnet": run_devnet,
+        "testnet": "testnet",
+        "integration": "https://external.integration.starknet.io",
+    }
+
+    return net_address[net]
+
+
+@pytest.fixture(
+    params=[
+        "typed_data_example.json",
+        "typed_data_felt_array_example.json",
+        "typed_data_long_string_example.json",
+        "typed_data_struct_array_example.json",
+    ],
+)
+def typed_data(request) -> TypedData:
+    """
+    Returns TypedData dictionary example
+    """
+    file_name = getattr(request, "param")
+    file_path = TYPED_DATA_DIR / file_name
+
+    with open(file_path, "r", encoding="utf-8") as file:
+        typed_data = json.load(file)
+
+    return typed_data
+
+
+@pytest_asyncio.fixture(scope="module")
+async def cairo_serializer(gateway_account_client: AccountClient) -> CairoSerializer:
+    """
+    Returns CairoSerializer for "simple_storage_with_event.cairo"
+    """
+    client = gateway_account_client
+    contract_content = (CONTRACTS_DIR / "simple_storage_with_event.cairo").read_text(
+        "utf-8"
+    )
+
+    deployment_result = await Contract.deploy(
+        client, compilation_source=contract_content
+    )
+    await deployment_result.wait_for_acceptance()
+    contract = deployment_result.deployed_contract
+
+    return CairoSerializer(identifier_manager=contract.data.identifier_manager)
```

Comment:
I think the fixtures in this module are not that related - I think it would make sense to rename this file to `misc` or `miscellaneous` and perhaps we should consider moving devnet related things to separate file like `devnet`.